packages/spearly-cms-js-core/src/spec/Generator-List.spec.ts

```diff
@@ -9,37 +9,71 @@ const convertTestData = [
-        ], 2),
-        expected: "<h1>title</h1><div>description</div><h1>title</h1><div>description</div>"
+        mockData: generateServerListFromContent([
+            generateServerContent([
+                {identifier: "title", inputType: "text", value: "title" },
+                {identifier: "description", inputType: "text", value: "description" },
+            ], 'alias-1'),
+            generateServerContent([
+                {identifier: "title", inputType: "text", value: "title2" },
+                {identifier: "description", inputType: "text", value: "description2" },
+            ], 'alias-2'),
+        ]),
+        expected: "<h1>title</h1><div>description</div><h1>title2</h1><div>description2</div>"
+    },
+    { 
+        testName: "cms-loop reference sub loop",
+        template: `<h1>{%= blog_title %}</h1><div cms-loop cms-field="ref">{%= blog_ref_author %}</div>`,
+        options: {} as unknown as SpearlyJSGeneratorOption,
+        apiOptions: new Map<string, string>(),
+        contentType: "blog",
+        mockData: generateServerListFromContent([
+            generateServerContent([
+                {identifier: "title", inputType: "text", value: "title1" },
+                {identifier: "ref", inputType: "content_type", value: {
+                    data: [
+                        generateServerContent([
+                            {identifier: "author", inputType: "text", value: "ref-title1"}
+                        ], 'alias-sub-1')
+                    ]
+                }}
+            ], 'alias-1'),
+            generateServerContent([
+                {identifier: "title", inputType: "text", value: "title2" },
+                {identifier: "ref", inputType: "content_type", value: {
+                    data: [
+                        generateServerContent([
+                            {identifier: "author", inputType: "text", value: "ref-title2"}
+                        ], 'alias-sub-2')
+                    ]
+                }}
+            ], 'alias-2'),
+        ]),
+        expected: "<h1>title1</h1><div>ref-title1</div><h1>title2</h1><div>ref-title2</div>"
     }
 ]
 
-describe('SpearlyJSGenerator', () => {
-    describe('generateContent-list: リスト生成', () => {        
-        convertTestData.forEach(testData => {
-            let generator = new SpearlyJSGenerator('apikey', 'domain', 'analyticsDomain', testData.options)
-            it(`generateContent: ${testData.testName}`, async () => {
-                // モック
-                Object.defineProperty(generator, 'client', {
-                    value: {
-                        getList: (_: string, param: GetParams | undefined) => {
-                            return Promise.resolve(testData.mockData)
-                        }
+describe('generateList: リスト生成', () => {        
+    convertTestData.forEach(testData => {
+        let generator = new SpearlyJSGenerator('apikey', 'domain', 'analyticsDomain', testData.options)
```

Comment:
再代入がなさそうだったのでconstで良さそうですかね？

```suggestion
        const generator = new SpearlyJSGenerator('apikey', 'domain', 'analyticsDomain', testData.options)
```